app/main.py

```diff
@@ -1,14 +1,69 @@
-class Validator:
-    pass
+from __future__ import annotations
+from abc import ABC, abstractmethod
+from typing import Any
 
 
-class Number:
-    pass
+class Validator(ABC):
+    def __set_name__(self, owner: Validator, name: str) -> None:
+        self.protected_name = "_" + name
 
+    def __get__(self, instance: Number | OneOf, owner: Validator) -> int | str:
+        return getattr(instance, self.protected_name)
 
-class OneOf:
-    pass
+    def __set__(self, instance: Number | OneOf, value: Any) -> None:
```

Comment:
In the future, do not use your child classes in the type annotation for this method. The parent class should not be aware of its child classes at this stage. It would be better to provide a `Validator` class for the type hinting here.